gdnative-core/src/vector2.rs

```diff
@@ -15,6 +14,156 @@ impl FromVariant for Vector2 {
+    /// and returns the result at position `t`. `t` is in the range of 0.0 - 1.0, representing
+    /// the amount of interpolation.
+    fn cubic_interpolate(self, b: Self, pre_a: Self, post_b: Self, t: f32) -> Self;
+    /// Returns the normalized vector pointing from this vector to `point`.
+    fn direction_to(self, point: Point2) -> Self;
+    /// Returns the distance to `point`.
+    fn distance_to(self, point: Point2) -> Length;
+    /// Returns the squared distance to `point`. Prefer this function over distance_to if you
+    /// need to sort vectors or need the squared distance for some formula.
+    fn distance_squared_to(self, point: Point2) -> Length;
+    /// Returns the vector projected onto the `other` vector.
+    fn project(self, other: Self) -> Self;
+    /// Returns the vector reflected from a plane defined by the given `normal`.
+    fn reflect(self, normal: Self) -> Self;
+    /// Returns the vector rotated by `angle` radians.
+    fn rotated(self, angle: Angle) -> Self;
+    /// Returns the component of the vector along a plane defined by the given normal.
+    fn slide(self, normal: Self) -> Self;
+    /// Returns the vector snapped to a grid with the given size.
+    fn snapped(self, by: Self) -> Self;
+    /// Returns a perpendicular vector.
+    fn tangent(self) -> Self;
+}
+
+impl Vector2Godot for Vector2 {
+    #[inline]
+    fn angle_to(self, other: Self) -> Angle {
+        Angle::radians(Trig::fast_atan2(self.cross(other), self.dot(other)))
+    }
+
+    #[inline]
+    fn angle_to_point(self, point: Point2) -> Angle {
+        Angle::radians(Trig::fast_atan2(self.y - point.y, self.x - point.x))
+    }
+
+    #[inline]
+    fn aspect(self) -> f32 {
+        self.x / self.y
+    }
+
+    #[inline]
+    fn bounce(self, normal: Self) -> Self {
+        let normal = normal.normalize();
+        normal * ((-2.0) * self.dot(normal)) + self
+    }
+
+    #[inline]
+    fn clamped(self, length: Length) -> Self {
+        if self.length() > length.get() {
+            self.normalize() * length.get()
+        } else {
+            self
+        }
+    }
+
+    #[inline]
+    fn cubic_interpolate(self, b: Self, pre_a: Self, post_b: Self, t: f32) -> Self {
+        let v0 = pre_a;
+        let v1 = self;
+        let v2 = b;
+        let v3 = post_b;
+
+        let t2 = t * t;
+        let t3 = t2 * t;
+
+        ((v1 * 2.0)
+            + (-v0 + v2) * t
+            + (v0 * 2.0 - v1 * 5.0 + v2 * 4.0 - v3) * t2
+            + (-v0 + v1 * 3.0 - v2 * 3.0 + v3) * t3)
+            * 0.5
+    }
+
+    #[inline]
+    fn direction_to(self, point: Point2) -> Self {
+        (point - self).to_vector().normalize()
+    }
+
+    #[inline]
+    fn distance_to(self, point: Point2) -> Length {
+        let squared = (self.x - point.x).powi(2) + (self.y - point.y).powi(2);
+        Length::new(squared.sqrt())
+    }
+
+    #[inline]
+    fn distance_squared_to(self, point: Point2) -> Length {
```

Comment:
This should be a method of Point2 (not only something that takes a point in parameter).
It can be simply implemented as (point - *self).square_length() by the way.